modules/component-store/src/component-store.ts

```diff
@@ -196,46 +206,34 @@ export class ComponentStore<T extends object> implements OnDestroy {
   /**
    * Creates a selector.
    *
-   * This supports combining up to 4 selectors. More could be added as needed.
-   *
    * @param projector A pure projection function that takes the current state and
    *   returns some new slice/projection of that state.
    * @param config SelectConfig that changes the behavior of selector, including
    *   the debouncing of the values until the state is settled.
    * @return An observable of the projector results.
    */
-  select<R>(projector: (s: T) => R, config?: SelectConfig): Observable<R>;
-  select<R, S1>(
-    s1: Observable<S1>,
-    projector: (s1: S1) => R,
-    config?: SelectConfig
-  ): Observable<R>;
-  select<R, S1, S2>(
-    s1: Observable<S1>,
-    s2: Observable<S2>,
-    projector: (s1: S1, s2: S2) => R,
-    config?: SelectConfig
-  ): Observable<R>;
-  select<R, S1, S2, S3>(
-    s1: Observable<S1>,
-    s2: Observable<S2>,
-    s3: Observable<S3>,
-    projector: (s1: S1, s2: S2, s3: S3) => R,
-    config?: SelectConfig
-  ): Observable<R>;
-  select<R, S1, S2, S3, S4>(
-    s1: Observable<S1>,
-    s2: Observable<S2>,
-    s3: Observable<S3>,
-    s4: Observable<S4>,
-    projector: (s1: S1, s2: S2, s3: S3, s4: S4) => R,
+  select<Result>(
+    projector: (s: T) => Result,
     config?: SelectConfig
-  ): Observable<R>;
+  ): Observable<Result>;
+  select<Selectors extends Observable<unknown>[], Result>(
+    ...args: [
+      ...selectors: Selectors,
+      projector: SelectorProjector<Selectors, Result>
+    ]
+  ): Observable<Result>;
+  select<Selectors extends Observable<unknown>[], Result>(
+    ...args: [
+      ...selectors: Selectors,
+      projector: SelectorProjector<Selectors, Result>,
+      config: SelectConfig
+    ]
+  ): Observable<Result>;
   select<
-    O extends Array<Observable<unknown> | SelectConfig | ProjectorFn>,
-    R,
-    ProjectorFn = (...a: unknown[]) => R
-  >(...args: O): Observable<R> {
+    Selectors extends Array<Observable<unknown> | SelectConfig | ProjectorFn>,
+    Result,
+    ProjectorFn = (...a: unknown[]) => Result
+  >(...args: Selectors): Observable<Result> {
     const { observables, projector, config } = processSelectorArgs(args);
 
     let observable$: Observable<unknown>;
```

Comment:
We might be able to get this work now:
```suggestion
    let observable$: Observable<Result>;
```

and at L254 just `return observable$.pipe(`